### PR TITLE
19641: Improves flexibility and documentation of entity traversal with nulls, MINOR

### PR DESCRIPTION
**Pull request by howsohazard**

### CODE REVIEW

File: src/Amalgam/entity/EntityManipulation.cpp [hunk 1]
```diff
@@ -650,11 +650,17 @@ EvaluableNodeReference EntityManipulation::FlattenEntity(Interpreter *interprete
 {
 	EvaluableNodeManager *enm = interpreter->evaluableNodeManager;
 
+	//TODO 19641: add code and tests to handle null root, and update documentation
```
Comment (calebwherry) [inline src/Amalgam/entity/EntityManipulation.cpp:653]: Remove this todo?

File: src/Amalgam/evaluablenode/EvaluableNodeTreeFunctions.h [hunk 2]
```diff
@@ -58,35 +58,54 @@ EntityReferenceType TraverseToExistingEntityReferenceViaEvaluableNodeIDPath(Enti
...
 		return EntityReferenceType(container->GetContainedEntity(sid));
 	}
 
-	size_t last_ce = ocn.size();
-	for(size_t i = 0; i < last_ce; i++)
+	//if traversing, then it needs to be a list, otherwise not a valid entity
+	if(id_path->GetType() != ENT_LIST)
+		return EntityReferenceType(nullptr);
+
+	//always keep one to two locks active at once to walk down the entity containers
```
Comment (calebwherry) [inline src/Amalgam/evaluablenode/EvaluableNodeTreeFunctions.h:84]: Not sure I understand this comment. What does "keep one to two locks active" mean in this context?

**Review by calebwherry — APPROVED** [review]
